### ISSUE TITLE
Some bug with AO inherited tables

### ISSUE THREAD

**Issue by jianlirong** [greenplum-db/gpdb#6068]
It seems currently, there is some problem with Appendonly inherited tables. More details are as follows.

### Greenplum version or build
The latest master branch on github

### OS version and uname -a
CentOS 7.2

### Expected behaviour
If an inherited table is not supported by appendonly tables, then a clear message should be sent to the client when defining the AO inherited table, rather than error out in the following DML operations.

### Actual behaviour
```
postgres=# delete from parent using todelete where parent.a = todelete.a;
ERROR:  append-only table "child" file segment "-1" entry does not exist
```
### Step to reproduce the behaviour
```
create table todelete (a int) with (appendonly=true) distributed by (a);
create table parent (a int, b int, c int) with (appendonly=true) distributed by (a);
create table child (a int, b int, c int) inherits (parent) with (appendonly=true) distributed by (b);
insert into parent select g, g, g from generate_series(1,5) g;
insert into child select g, g, g from generate_series(6,10) g;
insert into todelete select generate_series(3,4);
delete from parent using todelete where parent.a = todelete.a;
```

**Comment by asimrp** [comment]
Thanks for including the crisp repro @jianlirong.  Looks like we consider only partition tables but not inherited tables when assigning AO segno in `InitPlan()`.  With asserts enabled, the repro steps lead to this assertion failure:
```
#4  0x00005571f7f123c5 in ExceptionalCondition (conditionName=0x5571f8218edb "!(found)", errorType=0x5571f8217fd9 "FailedAssertion", fileName=0x5571f8218009 "execMain.c", lineNumber=2496) at assert.c:46
#5  0x00005571f7b3334c in ResultRelInfoSetSegno (resultRelInfo=0x5571fa21f168, mapping=0x5571fa224968) at execMain.c:2496
#6  0x00005571f7b316fc in InitPlan (queryDesc=0x5571fa237d68, eflags=1) at execMain.c:1778
#7  0x00005571f7b2f41d in standard_ExecutorStart (queryDesc=0x5571fa237d68, eflags=1) at execMain.c:583
#8  0x00005571f7b2e7b0 in ExecutorStart (queryDesc=0x5571fa237d68, eflags=1) at execMain.c:254
#9  0x00005571f7a930b2 in ExplainOnePlan (plannedstmt=0x5571fa237c50, into=0x0, es=0x7ffd8ae7f660, queryString=0x5571f9f89160 "explain delete from parent using todelete where parent.a = todelete.a;", params=0x0, planduration=0x7ffd8ae7f5d0) at explain.c:579
#10 0x00005571f7a92c24 in ExplainOneQuery (query=0x5571fa108080, into=0x0, es=0x7ffd8ae7f660, queryString=0x5571f9f89160 "explain delete from parent using todelete where parent.a = todelete.a;", params=0x0) at explain.c:416
#11 0x00005571f7a92757 in ExplainQuery (stmt=0x5571f9f8a148, queryString=0x5571f9f89160 "explain delete from parent using todelete where parent.a = todelete.a;", params=0x0, dest=0x5571fa11ad38) at explain.c:251
#12 0x00005571f7d94718 in standard_ProcessUtility (parsetree=0x5571f9f8a148, queryString=0x5571f9f89160 "explain delete from parent using todelete where parent.a = todelete.a;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0, dest=0x5571fa11ad38, completionTag=0x7ffd8ae7f8c0 "") at utility.c:809
#13 0x00005571f7d93acd in ProcessUtility (parsetree=0x5571f9f8a148, queryString=0x5571f9f89160 "explain delete from parent using todelete where parent.a = todelete.a;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0, dest=0x5571fa11ad38, completionTag=0x7ffd8ae7f8c0 "") at utility.c:373
#14 0x00005571f7d926b1 in PortalRunUtility (portal=0x5571fa128fb0, utilityStmt=0x5571f9f8a148, isTopLevel=1 '\001', dest=0x5571fa11ad38, completionTag=0x7ffd8ae7f8c0 "") at pquery.c:1375
#15 0x00005571f7d923c3 in FillPortalStore (portal=0x5571fa128fb0, isTopLevel=1 '\001') at pquery.c:1245
#16 0x00005571f7d91ca2 in PortalRun (portal=0x5571fa128fb0, count=9223372036854775807, isTopLevel=1 '\001', dest=0x5571f9f8a5d0, altdest=0x5571f9f8a5d0, completionTag=0x7ffd8ae7fb20 "") at pquery.c:981
#17 0x00005571f7d896f5 in exec_simple_query (query_string=0x5571f9f89160 "explain delete from parent using todelete where parent.a = todelete.a;") at postgres.c:1796
#18 0x00005571f7d8ecc9 in PostgresMain (argc=1, argv=0x5571f9fa5ae0, dbname=0x5571f9fa5940 "postgres", username=0x5571f9fa5918 "apraveen") at postgres.c:5223
```

**Comment by hlinnaka** [comment]
Fixed by commit 4c222185f. Thanks for the report!
